tests/src/Template/Provider/BaseProviderTest.php

```diff
@@ -276,15 +292,25 @@ private static function createPackage(
         string $name,
         string $type = 'project-builder-template',
         string $prettyVersion = '1.0.0',
-    ): Package\Package {
+    ): Package\CompletePackage {
         $versionParser = new VersionParser();
 
-        $package = new Package\Package($name, $versionParser->normalize($prettyVersion), $prettyVersion);
+        $package = new Package\CompletePackage($name, $versionParser->normalize($prettyVersion), $prettyVersion);
         $package->setType($type);
 
         return $package;
     }
 
+    private static function createAbandonedPackage(
+        string $name,
+        bool|string $abandoned,
+    ): Package\CompletePackage {
+        $package = self::createPackage($name);
+        $package->setAbandoned($abandoned);
+
+        return $package;
+    }
+
```

Comment:
Let's not create an additional method for abandoned packages, but rather use the existing `createPackage()` method with an additional property `bool $abandoned = false`.

Reply:
Again, completely agree. I've refactored accordingly.